### PR TITLE
Clang tidy

### PR DESCRIPTION
**Pull request by KaiSzuttor**
added some more checks

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -524,7 +524,57 @@ if(WITH_CLANG_TIDY)
     set(
       CMAKE_CXX_CLANG_TIDY
       "${CLANG_TIDY_EXE}"
-      "-checks=*,clang-analyzer-alpha.*,-misc-unused-*,-android-*,-fuchsia-*,-google-*,-llvm-*,-modernize-*,-readability-*,-cppcoreguidelines-*,-clang-analyzer-deadcode*,-cert-*,-clang-analyzer-core.NullDereference,-clang-analyzer-core.DivideZero,-misc-misplaced-widening-cast,-misc-macro-parentheses, -clang-analyzer-security.FloatLoopCounter, -clang-analyzer-optin.mpi.MPI-Checker,-boost-use-to-string,-hicpp-*, modernize-use-nullptr, modernize-use-auto, modernize-pass-by-value, modernize-avoid-c-arrays, modernize-use-equals-default, modernize-deprecated-headers, readability-non-const-parameter"
+      "-checks=*,\
```
Comment (fweik) [inline CMakeLists.txt]: I think you can put the check list into a `.clang-tidy` file similar to the `.clag-format`, which is maybe more readable than to put it into cmake.

File: src/core/RuntimeErrorCollector.cpp [hunk 2]
```diff
@@ -98,7 +98,7 @@ int RuntimeErrorCollector::count() const {
   int totalMessages;
   const int numMessages = m_errors.size();
 
-  all_reduce(m_comm, numMessages, totalMessages, std::plus<int>());
+  all_reduce(m_comm, numMessages, totalMessages, std::plus<>());
```
Comment (fweik) [inline src/core/RuntimeErrorCollector.cpp]: Can you check that boost.mpi can still detect that this is a MPI build-in operation with this?

Comment (fweik) [reply src/core/RuntimeErrorCollector.cpp]: There are multiples like this.

**Review by fweik — CHANGES_REQUESTED** [review]

File: src/core/RuntimeErrorCollector.cpp (hunk 2 above)
Comment (KaiSzuttor) [reply src/core/RuntimeErrorCollector.cpp]: How can I check that?

Comment (fweik) [reply src/core/RuntimeErrorCollector.cpp]: Look at the implementation

Comment (fweik) [reply src/core/RuntimeErrorCollector.cpp]: I can look it up

Comment (fweik) [reply src/core/RuntimeErrorCollector.cpp]: Looking at https://github.com/boostorg/mpi/blob/develop/include/boost/mpi/operations.hpp#L218, `std::plus<void>` is not yet recognized...

Comment (fweik) [reply src/core/RuntimeErrorCollector.cpp]: FYI: https://github.com/boostorg/mpi/pull/83, but that of course does not help us now.

Comment (KaiSzuttor) [reply src/core/RuntimeErrorCollector.cpp]: does this has a significant impact on performance?

Comment (fweik) [reply src/core/RuntimeErrorCollector.cpp]: Yes

**Review by fweik — APPROVED** [review]

**Pull request merged**